docs/mp/server.adoc

```diff
@@ -0,0 +1,373 @@
+include::{rootdir}/includes/mp.adoc[]
+
+== Content
+
+- <<Overview, Overview>>
+- <<Maven-Coordinates, Maven Coordinates>>
+- <<Usage, Usage>>
+- <<API, API>>
+- <<Configuration, Configuration>>
+- <<Examples, Examples>>
+- <<Reference, Reference>>
+
+== Overview
+
+Helidon provides a MicroProfile server implementation (`io.helidon.microprofile.server`) that encapsulates the Helidon WebServer.
+
+== Maven-Coordinates
+
+To enable MicroProfile Server add the helidon-microprofile-core bundle dependency to your project's `pom.xml`
+(see xref:../about/managing-dependencies.adoc[Managing Dependencies]).
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.microprofile.bundles</groupId>
+    <artifactId>helidon-microprofile-core</artifactId>
+</dependency>
+----
+
+MicroProfile Server is already included in the bundle.
+
+If full control over the dependencies is required, and you want to minimize the quantity of the dependencies -
+`Helidon MicroProfile Server` should be used. In this case the following dependencies should be included in your
+project's `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.microprofile.server</groupId>
+    <artifactId>helidon-microprofile-server</artifactId>
+</dependency>
+----
+
+== Usage
+
+Helidon Microprofile Server is used to collect and deploy JAX-RS application. You can either instantiate the server directly
+as is done in the xref:../guides/quickstart.adoc[Helidon MP Quickstart example]
+or use its built-in `create`. It will create the `server` instance. Calling `start` method will run the server.
+
+[source,java]
+----
+Server server = Server.create().start();
+----
+
+Call the `stop` method to stop the server.
+
+[source,java]
+----
+Server server = Server.create().start();
+...
+server.stop();
+----
+
+== API
+
+include::{rootdir}/includes/config/io.helidon.microprofile.server.Server.adoc[]
+
+The following table provides a brief description of routing annotations, including its parameters. More information
+in <<Configuring a reactive route>> section.
+
+[cols="3,5", role="flex, sm10"]
+|===
+|Annotation |Description
+
+a|
+----
+@RoutingName(
+    value = ""
+    required = false
+)
+----
+| Binds a JAX-RS Application or Helidon Service to a specific (named) routing on `WebServer`.The routing should have
+a corresponding named socket configured on the WebServer to run the routing on.
+
+a|
+----
+@RoutingPath("/path")
+----
+| Path of a Helidon Service to register with routing.
+
+|===
+
+== Configuration
+
+By default, the server uses the MicroProfile Config, but you may also want to use Helidon configuration.
+
+There are 3 default MicroProfile Config sources:
+
+* `System.getProperties()`
+* `System.getenv()`
+* all `META-INF/microprofile-config.properties` files on the class path
+* `application.yaml` on the classpath (read by default by Helidon Config)
+
+In this example, the configuration is in a file, and it includes Helidon configuration options.
+
+[source,properties]
```

Comment:
this section should be removed. include Server config and webserver config generated files as they cover all of this